RecoMuon/DetLayers/src/MuonME0DetLayerGeometryBuilder.cc

```diff
@@ -103,15 +106,17 @@ MuonME0DetLayerGeometryBuilder::buildLayer(int endcap,
     }
 
   }
-  LogTrace(metname) << "About to make a MuRingForwardLayer";
-  result = new MuRingForwardLayer(frontRings);
-  
-  LogTrace(metname) << "New MuRingForwardLayer with " << frontRings.size()
-                    << " and " << backRings.size()
-                    << " rings, at Z " << result->position().z()
-                    << " R1: " << result->specificSurface().innerRadius()
-                    << " R2: " << result->specificSurface().outerRadius();
   
+  LogTrace(metname) << "About to make a MuRingForwardLayer";
+  if(frontRings.size()!=0) result = new MuRingForwardLayer(frontRings);
```

Comment:
@jshlee: In answer to my original question, we found where the memory is deleted, so there is no problem:
https://github.com/cms-sw/cmssw/blob/CMSSW_8_1_0_pre4/RecoMuon/DetLayers/src/MuonDetLayerGeometry.cc#L33